pennylane/transforms/op_transforms.py

```diff
@@ -0,0 +1,504 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the op_transform decorator.
+"""
+# pylint: disable=protected-access
+import functools
+import inspect
+import os
+import warnings
+
+import pennylane as qml
+
+
+class OperationTransformError(Exception):
+    """Raised when there is an error with the op_transform logic"""
+
+
+class op_transform:
+    r"""Convert a function that applies to operators into a functional transform.
+
+    This allows the operator function to be used across PennyLane
+    on both instantiated operators as well as quantum functions.
+
+    By default, this decorator creates functional transforms that
+    accept a single operator. However, you can also register how the
+    transform acts on multiple operators. Once this is defined,
+    the transform can be used anywhere in PennyLane --- at the operator
+    level for operator arithmetic, or at the qfunc/QNode level.
+
+    .. warning::
+
+        This is an experimental feature, and is subject to change.

```

Comment:
@mariaschuld @antalszava I've completely re-written the docstring summary from scratch, let me know if this is better!